libraries/bezier-rs/src/subpath/solvers.rs

```diff
@@ -123,6 +124,118 @@ impl<ManipulatorGroupId: crate::Identifier> Subpath<ManipulatorGroupId> {
+		}
+		// If we can't draw the miter join, default to a bevel join
+		None
+	}
+
+	/// Returns the necessary information to create a round join with the provided center.
+	/// The returned items correspond to:
+	/// - The `out_handle` for the last manipulator group of `self`
+	/// - The new manipulator group to be added
+	/// - The `in_handle` for the first manipulator group of `other`
+	pub(crate) fn round_line_join(&self, other: &Subpath<ManipulatorGroupId>, center: DVec2) -> (DVec2, ManipulatorGroup<ManipulatorGroupId>, DVec2) {
+		let left = self.manipulator_groups[self.len() - 1].anchor;
+		let right = other.manipulator_groups[0].anchor;
+
+		let center_to_right = right - center;
+		let center_to_left = left - center;
+
+		let in_segment = self.get_segment(self.len_segments() - 1).unwrap();
+		let in_tangent = in_segment.tangent(TValue::Parametric(1.));
+
+		let mut angle = center_to_right.angle_between(center_to_left) / 2.;
+		let mut arc_point = center + DMat2::from_angle(angle).mul_vec2(center_to_right);
+
+		if (arc_point - left).angle_between(in_tangent).abs() > PI / 2. {
+			angle = angle - PI * (if angle < 0. { -1. } else { 1. });
+			arc_point = center + DMat2::from_angle(angle).mul_vec2(center_to_right);
+		}
+
+		let center_to_arc_point = arc_point - center;
+
+		// Based on https://pomax.github.io/bezierinfo/#circles_cubic
+		let handle_offset_factor = 4. / 3. * (angle / 4.).tan();
+
+		(
+			left - center_to_left.perp() * handle_offset_factor,
+			ManipulatorGroup::new(
+				arc_point,
+				Some(arc_point + center_to_arc_point.perp() * handle_offset_factor),
+				Some(arc_point - center_to_arc_point.perp() * handle_offset_factor),
+			),
+			right + center_to_right.perp() * handle_offset_factor,
+		)
+	}
+
+	/// Returns the necessary information to create a round cap between the end of `self` and the beginning of `other`.
+	/// The returned items correspond to:
+	/// - The `out_handle` for the last manipulator group of `self`
+	/// - The new manipulator group to be added
+	/// - The `in_handle` for the first manipulator group of `other`
+	pub(crate) fn round_cap(&self, other: &Subpath<ManipulatorGroupId>) -> (DVec2, ManipulatorGroup<ManipulatorGroupId>, DVec2) {
+		// Based on https://pomax.github.io/bezierinfo/#circles_cubic
+		const HANDLE_OFFSET_FACTOR: f64 = 0.551784777779014;
+
+		let left = self.manipulator_groups[self.len() - 1].anchor;
+		let right = other.manipulator_groups[0].anchor;
+
+		let center = (right + left) / 2.;
+		let center_to_right = right - center;
+		let center_to_left = left - center;
+
+		let arc_point = center + center_to_right.perp();
+		let center_to_arc_point = arc_point - center;
+
+		(
+			left - center_to_left.perp() * HANDLE_OFFSET_FACTOR,
+			ManipulatorGroup::new(
+				arc_point,
+				Some(arc_point + center_to_arc_point.perp() * HANDLE_OFFSET_FACTOR),
+				Some(arc_point - center_to_arc_point.perp() * HANDLE_OFFSET_FACTOR),
+			),
+			right + center_to_right.perp() * HANDLE_OFFSET_FACTOR,
```

Comment:
Could we eliminate some of the duplication between  `round_line_join(...)` and `round_cap(...)` with a helper functions that builds `left`, `right` and  the manipulator group using the `arc_point`, `center_to_arc_point` and `handle_offset_factor` arguments?